proposal/forward-with.md

```diff
@@ -0,0 +1,162 @@
+similar to `@use ... with`, but also allows additional `!default` flags,
+similar to a [variable declaration][]. Unlike `@use ... with`, unconfigured
+origin variables, and variables configured with a `!default` flag will remain configurable by any file importing the combined module. For example:
+
+[variable declaration]: https://github.com/sass/sass/blob/master/spec/variables.md#syntax
+
+```scss
+// _origin.scss
+$hue: 0 !default;
+$saturation: 50% !default;
+```
+
+```scss
+// _middleware.scss
+@forward "origin" with (
+  $hue: 330 !default, // Can be overridden by importing users.
+  $saturation: 70% // Cannot be overridden by importing users.
+);
+```
+
+```scss
+// entrypoint.scss
+@use "middleware" with (
+  $hue: 120 // override both the origin & middleware !default values
+);
+
+// middleware.$hue == 120
+// middleware.$saturation == 70%
+```
+
+Keyword arguments in the configuration must reference variable names as
+defined in the forwarded module, regardless of any concurent `as` clause:
+
+```scss
+// _origin.scss
+$hue: 0 !default;
+$color-hex: #ccc !default;
+```
+
+```scss
+// _middleware.scss
+@forward "origin" as color-* with (
+  $hue: 330, // the color-* prefix is not referenced in configuration
+  $color-hex: #966
+);
+```
+
+```scss
+// entrypoint.scss
+@use "middleware" as m;
+// m.$color-hue == 330
+// m.$color-hex == #966
+```
+
+A `@forward` rule configuration is applied to the source module even if the
+forwarding module acts as an entrypoint:
+
+```scss
+// _origin.scss
+$hue: 0 !default;
+```
+
+```scss
+// entrypoint.scss
+@forward "origin" with (
+  $hue: 330 !default
+);
+
+@use "origin"; // origin.$hue == 330
+```
+
+Multiple configurations can be chained in a single cascading "thread" that
+contains zero or more `@forward` rules, and zero or one terminal `@use` rule.
+Variables remain open to configuration in the chain as long as every mention
+includes the `!default` flag. Multiple threads configuring a single module will
+cause an error, even if they originate in the same file.
+
+## Syntax
+
+The new `WithClause` extends `@forward` to the follow grammar:
+
+<x><pre>
+**ForwardRule**     ::= '@forward' QuotedString AsClause? (ShowClause | HideClause)?  WithClause?
+**WithClause**      ::= 'with' '('
+&#32;                     KeywordArgument (',' KeywordArgument)\* ','?
+&#32;                   ')'
+**KeywordArgument** ::= '$' Identifier ':' Expression ('!default')
+</pre></x>
+
+## Semantics
+
+The `@forward ... with` semantics builds on the existing proposal for
+[Forwarding Modules][]. This proposal inserts some additional detail to the
+final step of the process:
+
+[Forwarding Modules]: ../accepted/module-system.md#forwarding-modules
+
+> Previously: "Otherwise, add `member` to `module` with the name `name`."
+
+* Otherwise, if `rule` has a `withClause` `with`:
+
+ * If `file` contains an `@use` rule with the same target URL and another
+   `withClause`, throw an error.
```

Comment:
Is this necessary? Won't [Loading Modules](https://github.com/sass/sass/blob/master/accepted/module-system.md#loading-modules) throw an error anyway if a module is loaded with configuration multiple times?